website/static/js/filepage/index.js

```diff
@@ -53,7 +53,15 @@ var SharePopover =  {
                         m('li', m('a[href="#embed"][data-toggle="tab"]', 'Embed'))
                     ]), m('br'),
                     m('.tab-content', [
-                        m('.tab-pane.active#share', m('.input-group', [
+                        m('.tab-pane.active#share', {
+                            config: function(element) {
+                                if ($(element).find('.shareButtons').length === 0) {
+                                    var shareContents = $('#shareContentHolder').html();
```

Comment:
This would allow you to reuse the component on both the file view page and the project overview.

```js
// in js/components/socialShare.js
var ShareDropdown = {
    view: function(ctrl, options) {
        return m('.btn-group dropdown', {}, [
            //...
            m.component(ShareButtons)
        ])
    }
}
```
```mako
// project.mako
<div id="shareDropdown"></div>
```

// project-dashboard-page.js
if (window.contextVars.node.isPublic) { m.mount(document.getElementById('shareDropdown'), ShareDropDown); }
```